Tests/Shared.Specs/GenericReadOnlyDictionaryAssertionSpecs.cs

```diff
@@ -0,0 +1,3031 @@
+﻿using System;
```

Comment:
This file seems to duplicate nearly all tests from `GenericDictionaryAssertions`.
I'm wondering what the purpose is?

Reply:
I was also not sure whether to "duplicate" all the tests. In general the older file tests dictionaries while the newer one tests read only dictionaries. 

I wanted full coverage of both scenarios, so I basically copied all the tests. They were extremely useful during the development of the PR, now a little bit less I guess. I would err on the side on leaving them, but will not protest if your consensus would go the other way.